Sources/Kitura/CodableRouter.swift

```diff
@@ -624,13 +637,41 @@ public struct CodableHelpers {
             return nil
         }
         do {
-            return try request.read(as: InputType.self)
+            return CodableHelpers.isContentTypeURLEncoded(request) ? try CodableHelpers.parseURLEncoded(InputType.self, from: request, response: response) : try request.read(as: InputType.self)
```

Comment:
We're evaluating Content-Type a second time, and each time we call `isContentTypeX()` we have to perform a Dictionary lookup and a String compare.  There will be a performance impact for existing JSON encoding here, because we first have to ask `isContentTypeURLEncoded`, before executing the existing `request.read`.
How about we evaluate the Content-Type once, and store the result alongside the request (eg. in an enum)? Then we could write the above something like:
```swift
switch request.contentType {
    case urlEncoded: return try CodableHelpers.parseURLEncoded(InputType.self, from: request, response: response)
    case jsonEncoded: return try request.read(as: InputType.self)
}
```
Checking the value of an enum on each request would be trivial extra overhead.

In addition, the guard above could be simplified to
```swift
guard request.contentType != .unsupported else {
```